model/host/host.go

```diff
@@ -435,35 +435,41 @@ func IsIntentHostId(id string) bool {
 	return strings.HasPrefix(id, "evg-")
 }
 
-func (h *Host) SetStatus(status, user string, logs string) error {
-	if h.Status == status {
+// SetStatus updates a host's status on behalf of the given user.
+func (h *Host) SetStatus(newStatus, user, logs string) error {
+	return h.SetStatusAndFields(newStatus, bson.M{}, user, logs)
+}
+
+// SetStatusAndFields is the same as SetStatus but also sets any of the other
+// given fields.
+func (h *Host) SetStatusAndFields(newStatus string, setFields bson.M, user, logs string) error {
+	if h.Status == newStatus {
 		return nil
 	}
 	if h.Status == evergreen.HostTerminated && h.Provider != evergreen.ProviderNameStatic {
 		msg := ErrorHostAlreadyTerminated
 		grip.Warning(message.Fields{
 			"message": msg,
 			"host_id": h.Id,
-			"status":  status,
+			"status":  newStatus,
 		})
 		return errors.New(msg)
 	}
 
+	setFields[StatusKey] = newStatus
```

Comment:
would this panic if someone passed in nil for setFields? If so, should we check nil?

Reply:
It would panic, but I don't really think it should be necessary to always double-check non-nilness in function parameters (e.g. we assume people don't pass nil maps typically) unless there's good reason to believe that the nil value has some special case meaning.